packages/lib-sourcify/src/SourcifyChain/SourcifyChain.ts

```diff
@@ -149,92 +240,76 @@ export class SourcifyChain {
     });
 
   getTx = async (creatorTxHash: string) => {
-    // Try sequentially all providers
-    for (const provider of this.providers) {
+    return this.executeWithCircuitBreaker(async (rpc) => {
+      if (!rpc.provider) {
+        throw new Error('No provider configured for this RPC');
+      }
+
+      logInfo('Fetching tx', {
+        creatorTxHash,
+        maskedProviderUrl: rpc.maskedUrl,
+      });
+      let tx: TransactionResponse | null;
       try {
-        logInfo('Fetching tx', {
-          creatorTxHash,
-          maskedProviderUrl: provider.maskedUrl,
-        });
-        // Race the RPC call with a timeout
-        const tx = await Promise.race([
-          provider.getTransaction(creatorTxHash),
-          this.rejectInMs(provider.maskedUrl),
+        tx = await Promise.race([
+          rpc.provider.getTransaction(creatorTxHash),
+          this.rejectInMs(rpc.maskedUrl),
         ]);
-        if (tx instanceof TransactionResponse) {
-          logInfo('Fetched tx', {
-            creatorTxHash,
-            maskedProviderUrl: provider.maskedUrl,
-          });
-          return tx;
-        } else {
-          throw new Error(
-            `Transaction ${creatorTxHash} not found on RPC ${provider.maskedUrl} and chain ${this.chainId}`,
-          );
-        }
       } catch (err) {
-        if (err instanceof Error) {
-          logWarn('Failed to fetch tx', {
-            creatorTxHash,
-            maskedProviderUrl: provider.maskedUrl,
-            chainId: this.chainId,
-            error: err.message,
-          });
-          continue;
-        } else {
-          throw err;
-        }
+        return { rpcFailed: true };
```

Comment:
I think here it makes sense to log the error or at least understand why the rpc is failing.

Reply:
There was already a log in `executeWithCircuitBreaker` but it is missing the error object. I decided to return the error instead of `rpcFailed` in order to add it in the existing log.